sdk/storage/storage-blob/MigrationGuide.md

```diff
@@ -0,0 +1,290 @@
+# Guide for migrating to `@azure/storage-blob` v12 from `azure-storage`
+
+This guide is intended to assist in the migration to `@azure/storage-blob` from the legacy `azure-storage` package. It will focus on side-by-side comparisons for similar operations between the two packages.
+
+We assume that you are familiar with `azure-storage`. If you are new to the Azure Storage Blob client library for JavaScript, please refer to the [README](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/storage/storage-blob/README.md) and [samples](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/storage/storage-blob/samples) rather than this guide.
+
+## Table of contents
+
+- [Migration benefits](#migration-benefits)
+  - [Cross Service SDK improvements](#cross-service-sdk-improvements)
```

Comment:
Typically we add two more sections under `Migration benefits` that would encourage people to upgrade
- New features
- Performance improvements

Do we have content for this? It need not be part of this PR, but it would be great to have it in a follow up. If you have ideas for this, please log an issue to track updating the migration guide

Reply:
I've opened an issue here to track it: https://github.com/Azure/azure-sdk-for-js/issues/19051